### PR TITLE
Web_Application

### PR DESCRIPTION
**Pull request by victor110298**

### CODE REVIEW

File: src/main/java/web_app/config/ApplicationConfig.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package web_app.config;
```
Comment (Taniann) [inline src/main/java/web_app/config/ApplicationConfig.java]: [Package names are all lowercase, with consecutive words simply concatenated together (no underscores)](https://mate-academy.github.io/style-guides/java/java.html#s5.2.1-package-names)

File: src/main/java/web_app/controller/ManufacturerController.java [hunk 2]
```diff
@@ -0,0 +1,4 @@
+package web_app.controller;
+
+public class ManufacturerController {
```
Comment (Taniann) [inline src/main/java/web_app/controller/ManufacturerController.java]: why do you need empty class?

File: src/main/java/web_app/entity/ProductEntity.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import static javax.persistence.GenerationType.IDENTITY;
+
+@Getter
+@Setter
+@Entity
+@NoArgsConstructor
+@AllArgsConstructor
+@Accessors(chain = true)
+@Table(name = "products")
+public class ProductEntity{
```
Comment (Taniann) [inline src/main/java/web_app/entity/ProductEntity.java]: formate your code

Comment (Taniann) [reply src/main/java/web_app/entity/ProductEntity.java]: add lombok annotation for equals and hashcode where it is needed

File: src/main/java/webapp/controller/ManufacturerController.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+            return new ResponseEntity<>(HttpStatus.OK);
+        }
+        return new ResponseEntity<>(entity, HttpStatus.NOT_FOUND);
+    }
+
+    @GetMapping(path = "/{id}")
+    @PreAuthorize("hasAnyRole('ADMIN', 'USER')")
+    public ResponseEntity<Manufacturer> getProducer(@PathVariable("id") Long id) {
+        Manufacturer manufacturer = (Manufacturer) manufacturerService.findOneById(id);
+        if (manufacturer == null) return new ResponseEntity<>(HttpStatus.NOT_FOUND);
```
Comment (Taniann) [inline src/main/java/webapp/controller/ManufacturerController.java]: add braces 
https://mate-academy.github.io/style-guides/java/java.html#s4.1.1-braces-always-used

File: src/main/java/webapp/controller/ManufacturerController.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        else
+            return new ResponseEntity<>(manufacturer, HttpStatus.OK);
```
Comment (Taniann) [inline src/main/java/webapp/controller/ManufacturerController.java]: the same

File: src/main/java/webapp/controller/UserController.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        }
+        return new ResponseEntity<>(user, HttpStatus.NOT_FOUND);
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<User> getUser(@PathVariable("id") Long id) {
+
+        User user = (User) service.findOneById(id);
+        if (user == null)
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
```
Comment (Taniann) [inline src/main/java/webapp/controller/UserController.java]: the same about braces

File: src/main/java/webapp/dto/Product.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+import java.util.UUID;
+
+@Getter
+@Setter
+@EqualsAndHashCode
+@NoArgsConstructor
+@AllArgsConstructor
+@Accessors(chain = true)
+public class Product extends ProductEntity {
+    private Long id;
```
Comment (Taniann) [inline src/main/java/webapp/dto/Product.java]: why do you need both id and uuid?

File: src/main/java/webapp/dto/Role.java [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+import java.util.UUID;
+
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@Accessors(chain = true)
+public class Role {
+    private Long id;
```
Comment (Taniann) [inline src/main/java/webapp/dto/Role.java]: the same

File: src/main/java/webapp/entity/RoleEntity.java [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+import java.util.UUID;
+
+import static javax.persistence.GenerationType.IDENTITY;
+
+@Getter
+@Setter
+@Entity
+@NoArgsConstructor
+@AllArgsConstructor
+
```
Comment (Taniann) [inline src/main/java/webapp/entity/RoleEntity.java]: remove redundant blank line

File: src/main/java/webapp/mapper/ProductMapper.java [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+
+@Component
+public class ProductMapper {
+    public Product mapProductEntityToProduct(ProductEntity entity){
+        return new Product()
```
Comment (Taniann) [inline src/main/java/webapp/mapper/ProductMapper.java]: use builders

File: src/main/java/webapp/mapper/UserMapper.java [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+import webapp.dto.User;
+import webapp.entity.UserEntity;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+
+@Component
+public class UserMapper {
```
Comment (Taniann) [inline src/main/java/webapp/mapper/UserMapper.java]: use builders in mapper classes

File: src/main/java/webapp/service/ManufacturerService.java [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+import webapp.mapper.ManufacturerMapper;
+import webapp.repository.ManufacturerRepository;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class ManufacturerService implements BaseService<ManufacturerEntity> {
+    private final ManufacturerRepository repository;
+    ManufacturerMapper mapper=new ManufacturerMapper();
```
Comment (Taniann) [inline src/main/java/webapp/service/ManufacturerService.java]: why dont you inject it?

File: src/main/java/webapp/dto/Product.java (hunk 7 above)
Comment (victor110298) [reply src/main/java/webapp/dto/Product.java]: Olexandr said that we need both of them

**Pull request merged**